packages/shim-deno/src/deno/stable/variables/std.ts

```diff
@@ -1,7 +1,91 @@
 ///<reference path="../lib.deno.d.ts" />
 
-import { File } from "../classes/FsFile.js";
+interface Deferred<T> extends Promise<T> {
+  resolve(value: T): void;
+  reject(reason?: any): void;
+}
+function defer<T>(): Deferred<T> {
+  const transit: any = {};
+  const result = new Promise((resolve, reject) =>
+    Object.assign(transit, { resolve, reject })
+  );
+  return Object.assign(result, transit);
+}
 
-export const stdin: typeof Deno.stdin = new File(0);
-export const stdout: typeof Deno.stdout = new File(1);
-export const stderr: typeof Deno.stderr = new File(2);
+export const stdin: typeof Deno.stdin = {
+  rid: 0,
+  read(p) {
+    const deferred = defer<number | null>();
+    p.fill(0);
+    process.stdin.resume();
+    process.stdin.once("readable", () => {
+      const data = process.stdin.read(p.length) ?? process.stdin.read();
```

Comment:
Could falling back to `read()` cause more data to be read than what's in the buffer and then for `p.set(data)` to fail below with a `RangeError`?

Reply:
I think no. 
> The optional size argument specifies a specific number of bytes to read. If size bytes are not available to be read, null will be returned unless the stream has ended, in which case, all of the data remaining in the internal buffer will be returned.
If the size argument is not specified, all of the data contained in the internal buffer will be returned.

According to the [documentation](https://nodejs.org/dist/latest-v17.x/docs/api/stream.html#readablereadsize), if buffer has data more than `p.length` bytes, it doesn't fall back to `stdin.read()`.
but buffer has data less than `p.length` bytes or is empty, `stdin.read()` will return all of the data contained in the buffer which is always less than `p.length` bytes.